### PR TITLE
Add API hipblasLtIsDeviceSupported() to query if current device is supported

### PR DESCRIPTION
**Pull request by jichangjichang**

### CODE REVIEW

**Comment by KKyang** [comment]
Why we need this?

**Comment by jichangjichang** [comment]
> Why we need this?
https://github.com/pytorch/pytorch/pull/128753
pytroch team request: pytroch doesn't have a way to check if current device is supported by hipblaslt or not. They don't want to add hardcode to check "gfx90a", "gfx942" in framework side.

**Comment by IMbackK** [comment]
this pr suffers from the same issue as this one here https://github.com/pytorch/pytorch/pull/128753#discussion_r1649620265 ie it is impossible to get here with a hipblaslt unsupported gpu without the rocm runtime haveing silently (or not) entered a failed state.

**Review by KKyang — APPROVED** [review]

**Comment by jithunnair-amd** [comment]
@jichangjichang I'd like us to have a discussion about whether this solution is the best way to go, before we merge this PR. It seems there will still be open issues at the framework level after this PR.

**Comment by jichangjichang** [comment]
> @jichangjichang I'd like us to have a discussion about whether this solution is the best way to go, before we merge this PR. It seems there will still be open issues at the framework level after this PR.

@jithunnair-amd  Any suggestions?

**Comment by jithunnair-amd** [comment]
@jichangjichang From offline discussions with @jeffdaily, he had concerns about this being a "hip" API when no "cuda" equivalent exists; suggestion being to make this only a "roc" api in accordance with convention.

File: library/src/amd_detail/rocblaslt/src/tensile_host.cpp [hunk 1]
```diff
@@ -2902,6 +2905,23 @@ std::string getSolutionNameFromAlgoIndex(rocblaslt_handle handle, const rocblasl
     return solution->solutionName;
 }
 
+rocblaslt_status isDeviceSupported()
```
Comment (jithunnair-amd) [inline library/src/amd_detail/rocblaslt/src/tensile_host.cpp:2908]: @jichangjichang Is this code thread-safe?

Comment (jichangjichang) [reply library/src/amd_detail/rocblaslt/src/tensile_host.cpp:2908]: @jithunnair-amd  yes

**Comment by jichangjichang** [comment]
> @jichangjichang From offline discussions with @jeffdaily, he had concerns about this being a "hip" API when no "cuda" equivalent exists; suggestion being to make this only a "roc" api in accordance with convention.

@jithunnair-amd hipblaslt already has many "none cuda" equivalent APIs. We use "**hipblasltExt**" prefix for this kinds of APIs. I can rename the query APIs as "hipblasLt**Ext**IsDeviceSupported()". How do you think?

**Comment by jithunnair-amd** [comment]
> > @jichangjichang From offline discussions with @jeffdaily, he had concerns about this being a "hip" API when no "cuda" equivalent exists; suggestion being to make this only a "roc" api in accordance with convention.
> 
> @jithunnair-amd hipblaslt already has many "none cuda" equivalent APIs. We use "**hipblasltExt**" prefix for this kinds of APIs. I can rename the query APIs as "hipblasLt**Ext**IsDeviceSupported()". How do you think?

Sounds okay to me if we have precedent. @jeffdaily, can you please let us know your opinion?

**Comment by jeffdaily** [comment]
I will make an exception for this.  hipblaslt doesn't have a public rocblaslt interface or library.  We really should be putting such extensions as "roc" APIs only.  The HIP interfaces for our libraries are meant to be a 1-1 correspondence with CUDA libraries.  Period.  We're adding more and more API surface area to the HIP part of hipblaslt that goes against our policies.  At least you're putting it behind an "ext" API name to avoid confusion here.

**Comment by jichangjichang** [comment]
No need so far, close first

**Pull request closed without merging**